src/main.cpp

```diff
@@ -27,24 +27,47 @@
 unsigned long startMillis;
 unsigned long currentMillis;
 
-//define your default values here, if there are different values in config.json, they are overwritten.
+// standard values ​​for the WiFi manager
+// !! some values ​​are overwritten by the WifiManager !!
+char wifi_AP[14] = "Energy2Shelly";
+String wifi_ssid;
+
 char input_type[40];
 char mqtt_server[80];
 char mqtt_port[6] = "1883";
 char mqtt_topic[60] = "tele/meter/SENSOR";
 char mqtt_user[40] = "";
 char mqtt_passwd[40] = "";
-char power_path[60] = "";
+// Energy values ​​input | example MQTT testvalue ->  {"ENERGY":{"Power": 9.99,"Consumption":77,"Production":33}}
+// Prerequisite, no change to the default values power_path energy_in_path energy_out_path
+char power_path[60] = "ENERGY.Power";
 char power_l1_path[60] = "";
 char power_l2_path[60] = "";
 char power_l3_path[60] = "";
-char energy_in_path[60] ="";
-char energy_out_path[60] ="";
+char energy_in_path[60] ="ENERGY.Consumption";
+char energy_out_path[60] ="ENERGY.Production";
+
+// Shelly emulated device configuration
+char shelly_app[7] = "Pro3EM";
+char shelly_gen[2] = "2";
+uint8_t shelly_slot = 1;
+char shelly_fw_id[32] = "20241011-114455/1.4.4-g6d2a586";
 char shelly_mac[13];
+char shelly_model[15] = "SPEM-003CEBEU";
 char shelly_name[26] = "shellypro3em-";
-char query_period[10] = "1000";
+char shelly_ver[7] = "1.4.4";
+char shelly_verBeta[12] = "1.4.9-beta6";
+char query_period[10] = "1000";   // milliseconds
+String power_variant = "monophase";
+
+uint8_t VALvoltage = 230;
+uint8_t VALfrequency = 50;
+uint8_t VALpowerFactor = 1;
 
 unsigned long period = 1000;
+unsigned long uptime = 0;         // uptime uC
+unsigned long secTick = 0;        // Zeit, zu der die Uhr zuletzt „tickte“

```

Comment:
bitte den Kommentar auch in Englisch halten

Reply:
ändere ich ab